scratch/misc/percent_flagged.R

```diff
@@ -0,0 +1,82 @@
+generate_flag_report <- function(df, df_index) {
+
+  # Extract the site and parameter from the df_name
+  site <- sub("^([^\\-]+)-.*$", "\\1", df_index)
+  parameter <- sub("^[^-]+-(.*)$", "\\1", df_index)
+
+  list_of_flags <- c("sonde not employed",
+                     "site visit",
+                     "sv window",
+                     "sensor specification range",
+                     "seasonal range",
+                     "slope flag suspect",
+                     "outside sd range",
+                     "repeated value",
+                     "missing data",
+                     "24hr anomaly",
+                     "anomaly window")
+
+  # summarize total data points
+  total_observations <- df %>%
+    summarise(n_total = n_distinct(DT_round)) %>%
+    pull(n_total)
+
+  # summarize total days
+  total_observations_dates <- df %>%
+    group_by(date = format(DT_round, "%m-%d-%Y")) %>%
```

Comment:
Suggestion: probably best to format dates as YYYY-MM-DD. KW might have thoughts on this, but generally it's a format that is autodetected. 

You can also use {lubridate} here:
`group_by(date = ymd(DT_round)) %>%`